bootstrap/resource_check_action.go

```diff
@@ -0,0 +1,48 @@
+package bootstrap
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/flynn/flynn/host/types"
+	"github.com/flynn/flynn/pkg/cluster"
+	"github.com/flynn/flynn/pkg/httphelper"
+)
+
+type ResourceCheckAction struct {
+	Ports []host.Port
+}
+
+func init() {
+	Register("resource-check", &ResourceCheckAction{})
+}
+
+func (a *ResourceCheckAction) Run(s *State) error {
+	conflicts := make(map[*cluster.Host][]host.Port)
+	for _, h := range s.Hosts {
+		req := host.ResourceCheck{Ports: a.Ports}
+		if err := h.ResourceCheck(req); err != nil {
+			if j, ok := err.(httphelper.JSONError); ok {
+				var resp host.ResourceCheck
+				if err := json.Unmarshal(j.Detail, &resp); err != nil {
+					return err
+				}
+				conflicts[h] = resp.Ports
+			} else {
+				return err
+			}
+		}
+	}
+	if len(conflicts) > 0 {
+		conflictMsg := "conflicts detected!\n\nThe following hosts have conflicting services listening on ports Flynn is configured to use:\n"
+		for host, ports := range conflicts {
+			conflictMsg += host.Addr() + ": "
```

Comment:
What do you think about stripping the port from the host IP? (`net.SplitHostPort`)